cloudinit/distros/__init__.py

```diff
@@ -217,23 +241,23 @@ def install_packages(self, pkglist):
                 pkg for pkg in uninstalled if pkg not in generic_packages
             }
             if failed:
-                LOG.error(error_message, failed)
+                LOG.info(error_message, failed)
             generic_packages = set(uninstalled)
 
         # Now attempt any specified package managers not explicitly supported
         # by distro
-        for manager, packages in packages_by_manager.items():
-            if manager.name in [p.name for p in self.package_managers]:
+        for manager_type, packages in packages_by_manager.items():
+            if manager_type.name in [p.name for p in self.package_managers]:
                 # We already installed/attempted these; don't try again
                 continue
             uninstalled.extend(
-                manager.from_config(self._runner, self._cfg).install_packages(
-                    pkglist=packages
-                )
+                manager_type.from_config(
+                    self._runner, self._cfg
+                ).install_packages(pkglist=packages)
             )
 
         if uninstalled:
-            LOG.error(error_message, uninstalled)
+            raise InstallerError(error_message, uninstalled)
```

Comment:
This needs to apply the formatting to the string
```suggestion
            raise InstallerError(error_message % uninstalled)
```